File: sample/src/main/java/com/haotsang/sample/theme/MyAdapter.java
package com.haotsang.sample.theme;


import androidx.annotation.NonNull;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseDataBindingHolder;
import com.haotsang.sample.R;
import com.haotsang.sample.databinding.ItemDatabindBinding;

public class MyAdapter extends BaseQuickAdapter<String, BaseDataBindingHolder<ItemDatabindBinding>> {

    public MyAdapter() {
        super(R.layout.item_databind);
    }

    @Override
    protected void convert(@NonNull BaseDataBindingHolder<ItemDatabindBinding> itemDatabindBindingBaseDataBindingHolder, String s) {
        ItemDatabindBinding dataBinding = itemDatabindBindingBaseDataBindingHolder.getDataBinding();
//        dataBinding.tvTitle.setTextColor(AppTheme.INSTANCE.getContent().getValue());
//        dataBinding.setColor(AppTheme.INSTANCE.getContent().getValue());
        dataBinding.tvTitle.setText(s);
        dataBinding.executePendingBindings();

    }
}

File: sample/src/main/java/com/haotsang/sample/module/theme/MyAdapter.kt
package com.haotsang.sample.module.theme

import android.content.Context
import androidx.lifecycle.LifecycleOwner

import com.haotsang.common.base.BaseBindingAdapter
import com.haotsang.common.base.BaseBindingViewHolder
import com.haotsang.sample.R
import com.haotsang.sample.databinding.ItemDatabindBinding


class MyAdapter(context: Context, lifecycle: LifecycleOwner) :
    BaseBindingAdapter<String, ItemDatabindBinding>(context, lifecycle) {

    override fun getLayout(viewType: Int): Int {
        return R.layout.item_databind
    }

    override fun onBindData(
        binding: ItemDatabindBinding,
        holder: BaseBindingViewHolder,
        data: String,
        position: Int
    ) {
        binding.tvTitle.text = data
    }

}

